src/Component.php

```diff
@@ -20,30 +20,39 @@ public function run(): void
             $client = new Client(['token' => $config->getToken(), 'url' => $config->getUrl()]);
                 $tokenInfo = $client->verifyToken();
             if (count($tokenInfo['bucketPermissions']) > 1) {
-                throw new UserException("The token has too broad permissions.");
+                throw new UserException('The token has too broad permissions.');
             }
             $bucket = array_keys($tokenInfo['bucketPermissions'])[0];
             if ($tokenInfo['bucketPermissions'][$bucket] !== 'write') {
-                throw new UserException("The token does not have write permissions to the bucket " . $bucket);
+                throw new UserException('The token does not have write permissions to the bucket ' . $bucket);
             }
             foreach ($config->getInputTables() as $table) {
-                $this->getLogger()->info("Processing table " . $table['destination']);
+                $this->getLogger()->info('Processing table ' . $table['destination']);
                 $manifest = $this->getManifestManager()->getTableManifest($table['destination']);
+                $primaryKey = $manifest['primary_key'] ?? [];
```

Comment:
a tady se přesně hodí `processor-create-manifest`, v případě, že bys chtěl použít jinej PK než je ve zdrojový tabulce (budeš například omezovat sloupce, nebo filtrovat na jednu hodnotu ve složeným PK).

což mě teda vede k tomu, že plánuju v UI mít možnost omezit sloupce, budu muset vyzkoušet, co to udělá s manifestem, resp. PK v manifestu, pokud odeberu sloupce, který jsou součástí PK

Reply:
ale jak nad tím přemýšlím, tak vlastně ten check, že PK nesedí, asi stačí. když zadefinuješ PK, který není ve sloupcích tabulky, tak zase zařve storage a to probublá do writeru jako user exception. writer by se měl prostě jen postarat, ať execne to, co má v configu a v manifestu, UI se musí postarat o zbytek, tj. UI musí pustit `processor-create-manifest` na nastavení PK, pokud ho uživatel jakkoliv zmodifikuje.

Reply:
ja nevim co mas furt s tim create manifest, manifest tam je prece vytvorenej z exportu a obsahuje PK tabulky. Jestli si odeberes sloupce, ktere jsou v PK, tak nemuzes prece nastavit jiny PK (nevis, ze bude vubec nastavitelny). Jediny, co by teda trosku davalo smysl je ten PK nenastavovat vubec (aby se pri vytvareni tabulky nenastavil na neexistujici sloupce)

Reply:
proč bych PK nemohl znát? můžu chtít zapisovat inkrementálně podle libovolnýho PK ze sloupců původní tabulky.

Reply:
nebo můžem říct, že je to hloupá věc a že nesmíš vybírat sloupce, měnit PK nebo filtrovat.

Reply:
tak mas tabulku:
id|name|text
1|test|prvni
2|test|druhy

PK je `id`, nastavis si filtr `text=prvni`, sloupec `text` prohlasis za PK, tabulku zapises, potom se ti v ni objevi:

id|name|text
1|test|prvni
2|test|druhy
3|prod|prvni

a melo by ti to pak zapsat jako co?

Podle me to nedava smysl, proste PK nemuzes "za behu" menit. Jak rikam, jediny, co mi smysl dava je mit moznost to PK nemit a ridit se tim, co je nastavene v cilove tabulce (nekdo si to nejak nastavil a ty mu tam jen sypes data) a co ty mas ve svym projektu nastaveny je uplne nezajimavy - cili jak to bylo predtim

Reply:
tak jsem to ještě řešil s padákem a resume je asi následující - PK se bude přebírat z tabulky (tj. z manifestu), sloupce se omezovat nebudou a když bude PK mismatch, tak to zařve (to bude znamenat, že někdo musel změnit tabulku v zdrojovým nebo cílovým projektu)

Reply:
takze to nehchavame jak to je

Reply:
jj, načíst a použít z manifestu, pokud nesedí, tak error

Reply:
no tak to je, takze merguju